### PR TITLE
Send Shopify header for employees

### PR DESCRIPTION
**Pull request by andyw8**
### WHAT is this pull request doing?

* We previously added behaviour to detect if someone is a Shopify employee. Now, for certain tasks, they will be prompted `Are you working on a 1P (1st Party) app?`. If they reply yes, this sends an additional header in the request which alters the behaviour in the partner app (that PR is currently in review).
* Not essential to the PR, but I've also added [minitest-fail-fast](https://github.com/teoljungberg/minitest-fail-fast) – it was useful when working to troubleshooting some kinds of noisy failures (it's inactive by default).
* See GitHub comments for further notes.

### CODE REVIEW

File: lib/shopify-cli/api.rb [hunk 1]
```diff
@@ -95,7 +95,9 @@ def current_sha
     def default_headers
       {
         'User-Agent' => "Shopify App CLI #{ShopifyCli::VERSION} #{current_sha} | #{ctx.uname}",
-      }.merge(auth_headers(token))
+      }.tap do |headers|
+        headers['X-Shopify-CLI-Employee'] = '1' if Shopifolk.acting_as_shopifolk?
```
Comment (andyw8) [inline lib/shopify-cli/api.rb:99]: Maybe `Cli` not `CLI` is the convention?

File: lib/shopify-cli/messages/messages.rb [hunk 2]
```diff
@@ -300,6 +310,8 @@ module Messages
               organization_not_found: "Cannot find a partner organization with that ID",
               partners_notice: "Please visit https://partners.shopify.com/ to create a partners account",
             },
+            first_party: "Are you working on a 1P (1st Party) app?",
```
Comment (andyw8) [inline lib/shopify-cli/messages/messages.rb:313]: @tiffanytse I made a small change to the wording here. Previously we agreed on `Are you building a 1P (1st Party) app?` but that doesn't make as much sense when using the `connect` command to connect to an existing app.

File: lib/shopify-cli/tasks/create_api_client.rb [hunk 3]
```diff
@@ -17,6 +17,15 @@ def call(ctx, org_id:, title:, type:)
           redir: [OAuth::REDIRECT_HOST]
         )
 
+        unless resp
+          ctx.abort("Error - empty response")
+        end
+
+        errors = resp.dig("errors")
```
Comment (andyw8) [inline lib/shopify-cli/tasks/create_api_client.rb:24]: While working on this I found that some kinds of errrors returned by Partners were not being handled correctly.

File: lib/shopify-cli/tasks/ensure_env.rb [hunk 4]
```diff
@@ -22,6 +22,9 @@ def call(ctx, regenerate: false, required: [:api_key, :secret])
       private
 
       def fetch_org
+        if Shopifolk.check && wants_to_run_against_shopify_org?
```
Comment (andyw8) [inline lib/shopify-cli/tasks/ensure_env.rb:25]: This impacts the following commands:

* Connect
* Scripts - Enable
* Scripts - Disable
* Scripts - Connect
* Populate Resource

File: lib/shopify-cli/tasks/select_org_and_shop.rb [hunk 5]
```diff
@@ -8,6 +8,9 @@ class SelectOrgAndShop < ShopifyCli::Task
       def call(ctx, organization_id: nil, shop_domain: nil)
         @ctx = ctx
         return response(organization_id.to_i, shop_domain) unless organization_id.nil? || shop_domain.nil?
+        if Shopifolk.check && wants_to_run_against_shopify_org?
```
Comment (andyw8) [inline lib/shopify-cli/tasks/select_org_and_shop.rb:11]: This impacts the following commands:

* Node - Forms - Create
* Rails - Forms - Create
* Populate Resource

File: lib/shopify-cli/tasks/create_api_client.rb (hunk 3 above)
Comment (paulomarg) [reply lib/shopify-cli/tasks/create_api_client.rb:24]: There are some edge cases where we get back the HTML error page from partners, I wonder if there is anything we can do about those.

File: test/shopify-cli/project_test.rb [hunk 6]
```diff
@@ -29,11 +29,13 @@ def test_current_fails_if_no_config
     end
 
     def test_write_writes_yaml
-      create_empty_config
       Shopifolk.stubs(:acting_as_shopify_organization?).returns(false)
+      Dir.stubs(:pwd).returns(@context.root)
+      FileUtils.touch(".shopify-cli.yml")
```
Comment (paulomarg) [inline test/shopify-cli/project_test.rb:34]: This is slightly curious. Is the fact that the test is touching the YAML file a problem, since the test is expected to write that file?

My concern is whether this would potentially mask an issue where we don't write the file if it doesn't exist.

**Review by paulomarg — APPROVED** [review]
LGTM! Added a couple of comments but probably nothing we need to action.

File: test/shopify-cli/project_test.rb (hunk 6 above)
Comment (andyw8) [reply test/shopify-cli/project_test.rb]: This wasn't needed, removed in 73ec07d4.

File: lib/shopify-cli/tasks/create_api_client.rb (hunk 3 above)
Comment (andyw8) [reply lib/shopify-cli/tasks/create_api_client.rb:24]: I think we can address that separately. There's already an assumption below that the response is a hash.

File: lib/project_types/node/commands/create.rb [hunk 7]
```diff
@@ -113,9 +113,20 @@ def build(name)
...
+
       def partners_endpoint
-        return 'https://partners.myshopify.io' if @ctx.getenv(ShopifyCli::PartnersAPI::LOCAL_DEBUG)
-        'https://partners.shopify.com'
+        domain = if @ctx.getenv(ShopifyCli::PartnersAPI::LOCAL_DEBUG)
+          'partners.myshopify.io'
+        else
+          'partners.shopify.com'
+        end
+        "https://#{domain}"
```
Comment (tanema) [inline lib/project_types/node/commands/create.rb:129]: Should these two methods actually be implemented in the PartnersAPI class?

**Review by tanema — APPROVED** [review]
Looks good!

File: lib/project_types/node/commands/create.rb (hunk 7 above)
Comment (andyw8) [reply lib/project_types/node/commands/create.rb:129]: Yeah, maybe, I see we have other methods in there already which return URLs. I'll try it and see.

Comment (andyw8) [reply lib/project_types/node/commands/create.rb]: f021ecc4

**Review by paulomarg — APPROVED** [review]
Moving these methods to the API class is really nice!
